heroic-component/src/main/java/com/spotify/heroic/Query.java

```diff
@@ -38,4 +38,6 @@
     private final Optional<QueryOptions> options;
     /* set of experimental features to enable */
     private final Optional<FeatureSet> features;
+    private Optional<QueryRequestMetadata> requestMetadata =
```

Comment:
Make `final`, and don't assign a default value. This will default to `Optional.empty()` if it is not specified.